src/main/perl/RuleBasedEditor.pm

```diff
@@ -1074,12 +1160,16 @@ sub _apply_rules
                 # remove_if_undef to be set.
                 # Note that this will never match instance parameters and will not remove entries
                 # no longer part of the configuration in a still existing LINE_VALUE_ARRAY or
-                # LINE_VALUE_STRING_HASH.
+                # LINE_VALUE_HASH.
                 unless ($attribute_present) {
                     if ($rule_parsing_options->{remove_if_undef}) {
                         $self->debug(1, "$function_name: attribute '$rule_info->{attribute}' undefined, ",
                                      "removing configuration line");
-                        $self->_commentConfigLine($keyword, $line_fmt);
+                        unless ( $self->_commentConfigLine($keyword, $line_fmt) ) {
+                            $self->error("$function_name: _commentConfigLine() encountered an internal error, ",
```

Comment:
if `_commentConfigLine` already reports an error, no need to repeat that here i think